controllers/storageautoscaler/storageautoscaler_controller.go

```diff
@@ -365,13 +318,219 @@ func (r *StorageAutoscalerReconciler) detectDuplicateStorageAutoScaler(ctx conte
 	return false, nil
 }
 
-func (r *StorageAutoscalerReconciler) detectLeanResourceProfile(storageCluster *ocsv1.StorageCluster) (bool, error) {
+func (r *StorageAutoscalerReconciler) detectLeanResourceProfile(ctx context.Context, storageCluster *ocsv1.StorageCluster, storageAutoScaler *ocsv1.StorageAutoScaler) (bool, error) {
 	if strings.ToLower(storageCluster.Spec.ResourceProfile) != "lean" {
 		return false, nil
 	}
 
 	err := fmt.Errorf("storage profile is lean, autoscaler does not support lean storage profile, delete the autoStorageScaler cr as scaling is not supported")
 	r.Log.Error(err, "storage profile is lean")
+
+	storageAutoScaler.Status.Error = &ocsv1.TimestampedError{
+		Message:   "storage profile is lean, autoscaler does not support lean storage profile, delete the autoStorageScaler cr as scaling is not supported",
+		Timestamp: metav1.Now(),
+	}
+	err = r.updateStatus(ctx, storageAutoScaler)
+	if err != nil {
+		return false, err
+	}
 	// not sending an error as user can see in the cr status
 	return true, nil
 }
+
+func (r *StorageAutoscalerReconciler) getCurrentOsdSizeAndCount(ctx context.Context, storageAutoScaler *ocsv1.StorageAutoScaler) (resource.Quantity, int, error) {
+	currentOsdCount := 0
+	currentOsdSize := resource.Quantity{}
+	osdCountQuery := `count by (device_class, namespace, managedBy) (ceph_osd_metadata)`
+	metrics, err := scrapeMetrics(ctx, r.OperatorNamespace, osdCountQuery, r.Log)
+	if err != nil {
+		r.Log.Error(err, "failed to scrape metrics")
+		return currentOsdSize, currentOsdCount, err
+	}
+
+	for _, osd := range metrics {
+		deviceClass := osd.Metric["device_class"]
+		verified := true
+		if string(deviceClass) == storageAutoScaler.Spec.DeviceClass {
+			currentOsdCount = int(osd.Value)
+			break
+		}
+
+		if !verified {
+			r.Log.Info("device class not found in metrics", "device class", storageAutoScaler.Spec.DeviceClass)
+			return currentOsdSize, currentOsdCount, errors.New("device class not found in metrics, device class: " + storageAutoScaler.Spec.DeviceClass)
```

Comment:
Similar suggestion in several other places too, but just mentioned here
```suggestion
			return currentOsdSize, currentOsdCount, errors.Errorf("device class not found in metrics, device class: %q", storageAutoScaler.Spec.DeviceClass)
```